### PR TITLE
doc: update developer notes with signet

### PR DESCRIPTION
**Pull request by jonatack**
Preview with working anchor links:

https://github.com/jonatack/bitcoin/blob/add-signet-to-developer-notes/doc/developer-notes.md

### CODE REVIEW

File: doc/developer-notes.md [hunk 1]
```diff
@@ -259,14 +259,15 @@ on all categories (and give you a very large `debug.log` file).
 The Qt code routes `qDebug()` output to `debug.log` under category "qt": run with `-debug=qt`
 to see it.
 
-### Testnet and Regtest modes
+### Signet, testnet, and regtest modes
```
Comment (MarcoFalke) [inline doc/developer-notes.md:262]:
```suggestion
### Test chains
```

nit?

**Review by maflcko — APPROVED** [review]
ACK ee701a92041c33f62b4e1adc3b10d2f1bb74f4c8

**Comment by 0xB10C** [comment]
ACK ee701a92041c33f62b4e1adc3b10d2f1bb74f4c8

**Comment by michaelfolkson** [comment]
ACK ee701a92041c33f62b4e1adc3b10d2f1bb74f4c8

File: doc/developer-notes.md (hunk 1 above)
Comment (jonatack) [reply doc/developer-notes.md:262]: thanks! no strong opinion about chains :chains: :chains: but reckoned @kallewoof and @ajtowns would relish seeing "Signet" in the section header :mega:

**Review by kallewoof — COMMENTED** [review]
Post-merge ACK

**Pull request merged**